### PR TITLE
[BugFix] Remove version and role file after failing to start fe at first time.

### LINKED ISSUE #39658: FE won't start forever if any exception occurs the first time FE starts.
Recently, we have noticed that if any exception occurs the first time FE starts, it can not start forever. The reason is that the FE will generate `VERSION` and `ROLE` files when the `GlobalStateMgr` is initializing but will not generate the `image` files like `image.0` which will be generated later. However, if any exception occurs between these two stages, the `image` files will not be generated but `VERSION` and `ROLE` files will exist.

The next time the FE starts, it reloads some information from the `VERSION` and `ROLE` files and skips to add the current frontend to `frontends` which could lead to `current node is not added to the cluster, will exit`, because the `image` files were not generated last time and it stops the FE from loading images and rebuilding the `frontends`. So, when the FE tries to transfer to the leader and check whether or not this node exists in the cluster, it never succeeds and always exits.

### Steps to reproduce the behavior (Required)

1. rewrite some code like below:

```java
// com.starrocks.server.GlobalStateMgr#initJournal
protected void initJournal() throws JournalException, InterruptedException {
    BlockingQueue<JournalTask> journalQueue =
            new ArrayBlockingQueue<JournalTask>(Config.metadata_journal_queue_size);
    if (Config.debug_throw_exception_in_init_journal) {
        // to produce an exception here.
        throw new RuntimeException("Throw exception when initialize journal.");
    }
    journal = JournalFactory.create(nodeMgr.getNodeName());
    journalWriter = new JournalWriter(journal, journalQueue);

    editLog = new EditLog(journalQueue);
}

// Config.java
@ConfField(mutable = true)
public static boolean debug_throw_exception_in_init_journal = true;
```

2. rebuild starrocks.
3. run `fe/bin/start_fe.sh --daemon`, it will exit and fe.log will display like this:

image

4. modify the config `debug_throw_exception_in_init_journal = false` in `fe.conf`
5. rerun `fe/bin/start_fe.sh --daemon`, it will exit and fe.log will display like this:

image

### Expected behavior (Required)

If an exception like UnknownHostException or IOException occurs, it probably recover from this exception soon, so we should ensure the FE could start successfully.

### Real behavior (Required)

Fail to start FE.

### StarRocks version (Required)

modified version based on branch `main`

### PR DESCRIPTION
**Pull request by wuxueyang96**
Why I'm doing:

What I'm doing:

Fixes #39658

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by miomiocat — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/server/NodeMgr.java [hunk 1]
```diff
@@ -218,6 +218,29 @@ public BrokerMgr getBrokerMgr() {
...
+            LOG.warn("Deleted role file {}, maybe because the firstly startup failed.", roleFile.getAbsolutePath());
+        } else {
+            LOG.warn("Failed to delete role file {}.", roleFile.getAbsolutePath());
+        }
+
+        File versionFile = new File(this.imageDir, Storage.VERSION_FILE);
+        if (versionFile.exists() && versionFile.delete()) {
+            LOG.warn("Deleted version file {}, maybe because the firstly startup failed.",
+                    versionFile.getAbsolutePath());
+        } else {
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/server/NodeMgr.java:239]: if versionFile.exists() is false , we do not need to print this warning log

File: fe/fe-core/src/main/java/com/starrocks/server/NodeMgr.java [hunk 2]
```diff
@@ -218,6 +218,28 @@ public BrokerMgr getBrokerMgr() {
...
+    public boolean isVersionAndRoleFilesNotExist() {
+        File roleFile = new File(this.imageDir, Storage.ROLE_FILE);
+        File versionFile = new File(this.imageDir, Storage.VERSION_FILE);
+        return !roleFile.exists() && !versionFile.exists();
+    }
+
+    private void removeMetaFileIfExist(String fileName) {
+        File file = new File(this.imageDir, fileName);
+        if (file.exists()) {
+            if (file.delete()) {
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/server/NodeMgr.java:230]: please catch 
SecurityException

**Review by caneGuy — COMMENTED** [review]
other LGTM

File: fe/fe-core/src/main/java/com/starrocks/StarRocksFE.java [hunk 3]
```diff
@@ -125,8 +125,24 @@ public static void start(String starRocksDir, String pidDir, String[] args) {
...
+            // must judge whether it is first time start here before initializing GlobalStateMgr.
+            boolean isFirstTimeStart = GlobalStateMgr.getCurrentState().getNodeMgr().isVersionAndRoleFilesNotExist();
+            try {
+                // init globalStateMgr.
+                // Possibly remove clusterId and role to ensure that the system is not left in a half-initialized state.
+                GlobalStateMgr.getCurrentState().initialize(args);
+            } catch (Exception e) {
+                try {
+                    if (isFirstTimeStart) {
+                        // If it is not the first time we start, we remove the cluster ID and role
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/StarRocksFE.java:137]: The comment should be 
// If it's the first time we start

File: fe/fe-core/src/main/java/com/starrocks/StarRocksFE.java [hunk 4]
```diff
@@ -125,8 +125,24 @@ public static void start(String starRocksDir, String pidDir, String[] args) {
             FrontendOptions.init(args);
             ExecuteEnv.setup();
 
-            // init globalStateMgr
-            GlobalStateMgr.getCurrentState().initialize(args);
+            // must judge whether it is first time start here before initializing GlobalStateMgr.
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/StarRocksFE.java:128]: Can you add these code into GlobalStateMgr to make the start function clean.

**Review by gengjun-git — DISMISSED** [review]

**Review by caneGuy — DISMISSED** [review]

**Comment by wuxueyang96** [comment]
I believe the failed unit tests are not relevant to this PR.

image

**Review by gengjun-git — DISMISSED** [review]

**Review by caneGuy — APPROVED** [review]

**Review by miomiocat — APPROVED** [review]

**Comment by nshangyiming** [comment]
ignore backport check: 2.5.19

**Pull request merged**